web/src/compose_setup.js

```diff
@@ -168,7 +169,14 @@ export function initialize() {
             const topic_name = $target.attr("data-topic-name");
 
             message_edit.with_first_message_id(stream_id, topic_name, (message_id) => {
-                message_edit.toggle_resolve_topic(message_id, topic_name, true);
+                if (!message_id) {
+                    // there is no message in the topic, so it is sufficient to
+                    // just remove the topic resolved prefix (✔) from the topic name
+                    const $input = $("input#stream_message_recipient_topic");
+                    $input.val(unresolve_name(topic_name));
```

Comment:
Does this trigger compose validation? I think you might need to call some `compose_recipient` module to change this properly.

Reply:
@timabbott  I didn't find any compose validation in the existing code, it just toggles the topic name and calls `channel.patch`. So all I did was mimic that, without the call to `channel.patch`.

Reply:
I made the following changes, mainly to address this issue:

```
diff --git a/web/src/compose_setup.js b/web/src/compose_setup.js
index bc9b91ff43..25c8fcb444 100644
--- a/web/src/compose_setup.js
+++ b/web/src/compose_setup.js
@@ -173,10 +173,15 @@ export function initialize() {
                     // There is no message in the topic, so it is sufficient to
                     // just remove the topic resolved prefix (✔) from the topic name.
                     const $input = $("input#stream_message_recipient_topic");
-                    $input.val(unresolve_name(topic_name));
+                    const new_topic = unresolve_name(topic_name);
+                    $input.val(new_topic);
                     // Trigger an input event, since this is a form of
                     // user-triggered edit to that field.
                     $input.trigger("input");
+
+                    // TODO: Probably this should also renarrow to the
+                    // new topic, if we were currently viewing the old
+                    // topic, just as if a message edit had occurred.
                 } else {
                     message_edit.toggle_resolve_topic(message_id, topic_name, true);
                 }
```